cluster/up.sh

```diff
@@ -33,5 +33,6 @@ for node in $(./cluster/kubectl.sh get nodes --no-headers | awk '{print $1}'); d
 done
 
 echo 'Deploying multus'
+# Taking latest multus (image is breathing and its on docker.io)
```

Comment:
As part of the effort to move images from docker.io to quay.io,
please suggest if we want to freeze multus yaml + the image it use,
currently it always take latest multus and latest stable image, which is on quay.

Reply:
I don't understand what you suggest here. This PR is to move to Quay only, how is freezing the yaml related?

Reply:
See please
https://github.com/kubevirt/ovs-cni/pull/150/files/deb0fb2539dcd2fc9b67dacd9b4e9389a8b1b094#diff-426e7a410d92c9ee2ea4e365697ede2f5f829539cdb3b10d623d170a326d6076R37

It fetches the latest yaml of multus, and this yaml has images which are on docker.io
docker.io/nfvpe/multus:stable
so it might be needed as part of this PR

one solution is to freeze the yaml, push the image to quay,
and patch the frozen yaml to point to the new quay image.

Reply:
Now I can understand what you mean. Let's copy the YAML, keep it under this project and point to KubeVirt's clone of Multus https://quay.io/repository/kubevirt/cluster-network-addon-multus?tab=tags.

Reply:
Sounds good for me, thanks

one point, since the latest yaml use `nfvpe/multus:stable`

I think that the easiest way to maintain what you just suggested, 
is to take the latest yaml , and push the image `nfvpe/multus:stable` to the quay you just pointed out,
instead taking the `latest` from there that exists, and trying to find the yaml that belong to it,

if its ok by you, i would need please credentials or help to push this image

Reply:
The quay is populated by CNAO automation. Please stick to the Multus version we currently use there. Let's keep this repository as stable as possible. Testing with the recent version of Multus is not a part of this PR.

Reply:
Correct me please if i am wrong,
i took a look on CNAO multus yaml, and its parametrized, its not the vanilla one
https://github.com/kubevirt/cluster-network-addons-operator/blob/3cc4d064a623104b13caccee879e9c0b9af1e97a/hack/components/bump-multus.sh
This file as part of the bump change it, and we need the original one that was fetched during the last bump.
In order to keep it simple to maintain, i think we should not reverse engineer the current already parameterized yaml, 
but don't touch this code on ovs-cni atm (we have a cache proxy on the ovs-cni prow lane, so we are safe).

The action item we can take, on other PR is to save also the vanilla yaml as part of the CNAO bump,
and even `docker cp` it to the image, then we can pull the image, copy it from the container image, and then apply it,
(it won't take extra time, as the deploy of the yaml does it anyhow),
alternatively, just save the raw yaml as part of the progress, so we can take it to the ovs-cni repo.

Otherwise, we will need to do a progress of reverse engineering a parameterized yaml again and again.

Please suggest what you think, thanks
Official raw latest
[1] https://raw.githubusercontent.com/intel/multus-cni/master/images/multus-daemonset.yml
Parameterized on CNAO
[2] https://github.com/kubevirt/cluster-network-addons-operator/blob/master/data/multus/001-multus.yaml

Reply:
> Correct me please if i am wrong,
> i took a look on CNAO multus yaml, and its parametrized, its not the vanilla one.
> 
> https://github.com/kubevirt/cluster-network-addons-operator/blob/3cc4d064a623104b13caccee879e9c0b9af1e97a/hack/components/bump-multus.sh

Both the [original](https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/deployments/Dockerfile) and [CNAO's](https://github.com/kubevirt/cluster-network-addons-operator/blob/3cc4d064a623104b13caccee879e9c0b9af1e97a/hack/components/bump-multus.sh#L117) Dockerfile build Multus from its sources and drop entrypoint script to the filesystem root. They should be fairly samiliar.

> This script as part of the bump change it, and we need the original one that was fetched during the last bump, (or to RE it, but i don't think its a good idea, please see below).
> In order to keep it simple to maintain, i think we should not reverse engineer the current already parameterized yaml,
> but don't touch this code on ovs-cni atm (we have a cache proxy on the ovs-cni prow lane, so we are safe).
> 
> The action item we can take, on other PR is to save also the vanilla yaml as part of the CNAO bump,
> and even `docker cp` it to the image, then we can pull the image, copy it from the container image, and then apply it,
> (it won't take extra time, as the deploy of the yaml does it anyhow),
> alternatively, just save the raw yaml as part of the progress, so we can take it to the ovs-cni repo.

CNAO does not care about the original manifest, it should not be concerned about needs of OVS CNI.

> 
> Otherwise, we will need to do a progress of reverse engineering a parameterized yaml again and again,
> structure changes, and its prone to mistakes, and time consuming, better to do it right imho.

Sorry, but I don't see the complexity you mention. The image CNAO builds is more or less the same as the one produced by Multus project. We can simply pin to a specific version of Multus kept under KubeVirt Quay and use the same yaml as we are using now (with changed IMAGE name) to use it. One image and one manifest with pinned versions we will probably not need to change whole year around.

> 
> Please suggest what you think, thanks
> Official raw latest
> [1] https://raw.githubusercontent.com/intel/multus-cni/master/images/multus-daemonset.yml
> Parameterized on CNAO
> [2] https://github.com/kubevirt/cluster-network-addons-operator/blob/master/data/multus/001-multus.yaml
> 
> > Testing with the recent version of Multus is not a part of this PR.
> 
> I understand, i just want to say please that this is already the case, i just strive to a method that would keep it maintainable and worth the effort.

Let's keep it simple.